apache/metron
# METRON-1675: Add Pagination to PCAP

## Contributor Comments
This PR is a subtask of the PCAP feature branch that adds pagination functionality to the PCAP search results. Unit tests are in place.

You can test by submitting a PCAP query and paging through the results. 

![pcap-paging](https://user-images.githubusercontent.com/7304869/43132105-04234caa-8f3b-11e8-823f-0df76b075554.gif)

## Testing
This has been tested in full dev. In order to test:

1. Spin up full dev.
1. Put pcap data inside of `/apps/metron/pcap/input` in your VM.
1. Start metron-alerts.
1. Sign in and click on the PCAP tab in the upper left hand corner of the screen.
1. Submit a pcap search which will return a set of results from your loaded pcap data, preferably on that will match at least a few dozen results.
1. Page through the results. The lines should update with the next/previous packets. When you download a file, the results should match the current page shown in the UI.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:


### For code changes:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```


### For documentation related changes:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.
